### DISCUSSION TITLE
Propagating connection across subdomains?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1608] [Developer Support]
Hey guys,
I've just implemented V2 into my project, muuuch better and cleaner ❤️ 
One question that I have.
Is there a possibility to propagate connection for WalletConnect from main domain abc.com to subdomain e.abc.com ?
So when user connects on main domain he doesn't have to redo wallet connection on subdomain 🙂

**Comment by xzilja** [comment]
There might be security implications here, but if you can bring over and restore `localStorage` on your subdomain, it should be possible i.e. via some sort of cookie tied to your main domain

**Reply by vildantursic** [reply]
That could work! I'll give it a try and inform you about the progress 😁

**Reply by vildantursic** [reply]
@GrahamMcBain @IljaDaderko 
First of all guys, Thank you!

Update:
I've tried this `localStorage` approach with **LedgerLive** and **MetaMask**.
With **LedgerLive** it works ✅ well.
But with **MetaMask** not at all ❌ . Data is there, and nothing is changing but it returns null for the account 😵‍💫 
I believe this would be MetaMask related issue, but I want to hear your opinion 😄 could there be some solution or I would have to cry on the MetaMask Github acc 😆
